README.md

```diff
@@ -1 +1,104 @@
-# oauth-flow-demo
\ No newline at end of file
+# oauth-flow-demo
+
+## System Design
+
+### Authorization
+
+Authorization is the process a client initiates to retrieve tokens for subsequent interactions with the application.
+
+```mermaid
+---
+title: Authorization
+---
+sequenceDiagram
+autonumber
+
+actor Client
+participant Authorization Service
+participant Authentication Service
+participant Token Issue Service
+
+
+Client-->Client: Generates code verifier and code challenge
+Client-->>Authorization Service: Calls the authorization endpoint with code challenge
+Note right of Client: GET /authorize
+Authorization Service-->Authorization Service: Encodes code challenge in state token
+Authorization Service-->>Authentication Service: Redirects to sign-in page with state token
+Authentication Service-->Authentication Service: Verifies user-supplied credentials
+Authentication Service-->>Authorization Service: Redirects to callback endpoint with state token
+Note left of Authentication Service: GET /callback
+Authorization Service-->Authorization Service: Decodes state token
+Authorization Service-->Authorization Service: Generates auth code
+Authorization Service-->Authorization Service: Stores auth code and code challenge
```

Comment:
If I'm being pedantic I might describe a shared data layer between `Authorization Service` and `Token Issue Service`, just because information this stores is verified in the Token Issuer

Reply:
Excellent suggestion